lib/puma/client.rb

```diff
@@ -40,6 +40,9 @@ class Client
     # no body share this one object since it has no state.
     EmptyBody = NullIO.new
 
+    # if compiled without ssl, MiniSSL doesn't exist
+    SSLError = HAS_SSL ? ::OpenSSL::SSL::SSLError : ::LocalJumpError
```

Comment:
I don't think LocalJumpError is appropriate, StandardError would be fine.

Reply:
Sorry for the poor explanation,  StandardError would catch everything.  What we want is to change the rescue (which is normally catching an ssl related error) to use an error that will never be raised, and do so only when Puma is not compiled with ssl.

Reply:
What about we define our own exception class that fulfill that requirement? We can give that class some good name that explains for the reader what's going on

Reply:
@dentarg

I'll look at that.  The rescue hocus-pocus is only needed in reactor.rb & server.rb (client.rb didn't need it, my mistake), and both involve rescuing MiniSSL::SSLError, so that could work.  Not sure where to put it.  AFK for a bit...

Reply:
A previous version of this relied on MiniSSL for detection, detection is now done with namespace children of MiniSSL.

Hence, `MiniSSL::SSLError` can now be defined in `Puma` when it is built without ssl support.  Hence, no need for 'hocus-pocus'.